SourceCode/Ray.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Ray.h"
+
+#include <cmath>
+
+Vector::Vector() : x(0), y(0), z(0){};
```

Comment:
I prefer the {} constructor. Here you're using integer literals to initialize floats, the {} will warn you about that.